scripts/encapp_tests.py

```diff
@@ -341,62 +390,111 @@ def build_tests(tests_json, device_model):
-                                                   output_dir,
-                                                   codec, res, mode,
-                                                   str(i_interval), str(br),
-                                                   str(duration),
-                                                   use_surface_enc)
-                                job_info_group.append(job_info)
-
-                            enc_jobs.add_job_info_group(job_info_group)
+                            for i_frame_size in i_frame_sizes:
+                                build_one_enc_job(workdir,
+                                                  device_model,
+                                                  enc_jobs,
+                                                  in_file,
+                                                  codec,
+                                                  res,
+                                                  mode,
+                                                  i_interval,
+                                                  i_frame_size,
+                                                  duration,
+                                                  group_desc,
+                                                  bitrates,
+                                                  use_surface_enc,
+                                                  input_format,
+                                                  input_res)
 
     return enc_jobs
 
 
-def run_encode_tests(tests_json, serial_no):
-
-    device_model, serial_no = check_device(serial_no)
+def run_encode_tests(tests_json, device_model, serial_no):
 
     # build tests
     enc_jobs = build_tests(tests_json, device_model)
 
     # run encode jobs
     enc_jobs.run_encodes(serial_no)
 
-    enc_jobs.get_rate_distortion()
+    enc_jobs.get_rate_distortion(serial_no)
 
-    rd_plot = RDPlot('Test RD Curve')
+    rd_plot = RDPlot()
     rd_plot.plot_rd_curve(enc_jobs.workdir+'/'+RD_RESULT_FILE_NAME)
 
 
+def check_device(serial_no):
+    # check for devices
+    adb_cmd = 'adb devices -l'
+    ret, stdout, stderr = run_cmd(adb_cmd)
+    if ret is False:
+        print('Failed to get adb devices')
+        sys.exit()
+    device_serials = []
+
+    for line in stdout.split('\n'):
+        if line == 'List of devices attached' or line == '':
+            continue
+        serial = line.split()[0]
+        device_serials.append(serial)
+
+    if len(device_serials) == 0:
+        print('No device connected')
+        exit(0)
+    elif len(device_serials) > 1:
+        if (serial_no is None):
+            print('More than one devices connected. \
+                   Please specifiy a device serial number')
+            exit(0)
+        elif (serial_no not in device_serials):
+            print('Specified serial number {} is invalid.'.format(serial_no))
+            exit(0)
+    else:
+        serial_no = serial
```

Comment:
I think the functionality here is wrong. If the user selects a `serial`, and it's different from the one she has connected, you should punt